pkg/service/agentservice.go

```diff
@@ -451,18 +464,11 @@ func (h *AgentHandler) selectWorkerWeightedByLoad(key workerKey, ignore map[*age
 		return nil, errors.New("no workers available")
 	}
 
-	normalizeLoad := func(load float32) int {
-		if load >= 1 {
-			return 0
-		}
-		return int((1 - load) * 100)
-	}
-
-	normalizedLoads := make(map[*agent.Worker]int)
-	var availableSum int
+	normalizedLoads := make(map[*agent.Worker]float32)
+	var availableSum float32
 	for _, w := range workers {
 		if _, ok := ignore[w]; !ok && w.Status() == livekit.WorkerStatus_WS_AVAILABLE {
-			normalizedLoads[w] = normalizeLoad(w.Load())
+			normalizedLoads[w] = max(0, 1-w.Load())
```

Comment:
This modification might be the cause of the issue we encountered. Can you please check?
[](https://github.com/livekit/agents/issues/884)
from agents, we implemented a load function / threshold that can return the value bigger than 1 for a long time. Recently, we've been encountering a lot of issues related to agents not receiving messages when rooms are created. We suspect this is the cause.